### PR TITLE
New Module: Listbox

### LINKED ISSUE #396: New Module: Listbox
This module will match the W3C Listbox pattern. This listbox pattern is similar to what we currently have in ebayui as the readonly combobox component. The readonly combobox component was recently converted from using input type=text to input type=button due to issues on iOS. However, this was only a stop gap solution. Once this Listbox module is added to skin, we will also add a listbox component to ebayui and revert the ebayui combobox back to using input type text for it's primary, non-readonly use case. We will also update the Skin combobox examples to be based more around the editable case.

### PR DESCRIPTION
**Pull request by ianmcburnie**
## Description

Added new Listbox module based on the [W3C Authoring Pattern](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html) (which we will also add to MIND patterns).

Added basic expand/collapse behaviour only to the website (I want to clean up & simplify the JS across DS4 and DS6 sites in a future PR).

## Context

Our readonly combobox has issues in iOS. This listbox module offers another alternative for creating a custom HTML select element.

The listbox borrows most of it's styling from existing "dropdown" styles, therefore this PR should be fairly straightforward. The listbox is styled exactly like the Menu module (makes sense, given that they both use an `expand-btn`). If we decide to make it more like the current Combobox, in terms of colours, then let's do that as a follow up PR as it may require changes to the button module (which I'd rather not do until the button PR is merged).

## References

Closes #396 

## Screenshots

Notice in the last two screenshots that there appears to be some issue with the Combobox overlapping the Listbox. I will look into this as part of #399.

screen shot 2018-10-08 at 12 34 07 pm
screen shot 2018-10-08 at 12 34 17 pm
screen shot 2018-10-08 at 12 34 21 pm
screen shot 2018-10-08 at 12 34 33 pm
screen shot 2018-10-08 at 12 34 38 pm
screen shot 2018-10-08 at 12 34 43 pm
screen shot 2018-10-08 at 12 35 06 pm
screen shot 2018-10-08 at 12 35 18 pm

### CODE REVIEW

File: docs/test/misc/ds6/alignment-controls-inline.html [hunk 1]
```diff
@@ -75,13 +97,37 @@ <h2>Medium</h2>
             </div>
         </div>
     </span>
+    <span class="listbox">
+        <button class="expand-btn expand-btn--medium" aria-expanded="false" aria-haspopup="listbox">
```
Comment (seangates) [inline docs/test/misc/ds6/alignment-controls-inline.html]: Looks like the `expand-btn--regular` needs some adjusting:

image

**Comment by seangates** [comment]
Otherwise, looks good!

**Comment by ianmcburnie** [comment]
Added missing padding property for `expand-btn--medium`. I see some duplicate rules for padding in the less file, but want to keep my changes to a minimum so I don't cause conflicts with your button PR. I also added the menu to the test page.

screen shot 2018-10-09 at 12 54 36 pm

screen shot 2018-10-09 at 12 54 28 pm

**Comment by seangates** [comment]
@ianmcburnie Well ... these files all change with the #401 PR.

**Pull request merged**